realm/realm-library/src/androidTest/java/io/realm/RealmAsyncQueryTests.java

```diff
@@ -301,7 +349,12 @@ public void execute(Realm realm) {
             @Override
             public void onSuccess() {
                 assertEquals(1, realm.where(AllTypes.class).count());
-                assertEquals(1, results.size());
+                // We cannot guarantee the async results get delivered from OS.
```

Comment:
So at this point
1. If the aynsc query has not been finished the first time, the `isLoad()` will return false. If user call `load()` manually then call any accessors on the results, it will do sync and return the new data.
2. if the async query has been returned before, but the newly refresh async queries has not been delivered, calling any accessor on the resultes, i will just sync the results and get the latest data.
3. if the async query has been returned and the query has been re-ran, the new results gets delivered already, then everything is good here.

It would be very difficult to make the async transaction perfect since all the timing sequence has highly fixed in the object store, but i think current behavior is acceptable from user perspective, since anyway he can always get the latest refreshed results in the async transaction callback.